Lib/fontbakery/profiles/name.py

```diff
@@ -261,39 +261,75 @@ def com_google_fonts_check_monospace(ttFont, glyph_metrics_stats):
-        familyname = familyname[0]
-
-        if not fullfontname.startswith(familyname):
-            yield FAIL,\
-                  Message("does-not",
-                          f"On the 'name' table, the full font name"
-                          f" (NameID {NameID.FULL_FONT_NAME}"
-                          f" - FULL_FONT_NAME: '{familyname}')"
-                          f" does not begin with font family name"
-                          f" (NameID {NameID.FONT_FAMILY_NAME}"
-                          f" - FONT_FAMILY_NAME: '{fullfontname}')")
-        else:
-            yield PASS, "Full font name begins with the font family name."
+    name_table = ttFont["name"]
+    names_compared = False  # Used for knowing if at least one comparison was attempted
+    passed = True
+
+    # Collect all the unique platformIDs, encodingIDs, and languageIDs
+    # used in the font's name table.
+    platform_ids = set(rec.platformID for rec in name_table.names)
+    encoding_ids = set(rec.platEncID for rec in name_table.names)
+    language_ids = set(rec.langID for rec in name_table.names)
+
+    # Now iterate over the platform/encoding/languageID sets and compare
+    # the full name and family name strings.
+    # The full name string is always taken from nameID 4, but
+    # the family name string can come from nameIDs 1, 16, or 21.
+    # We'll compare all of the ones contained in the font.
+    full_name_id = NameID.FULL_FONT_NAME
+    for plat_id in sorted(platform_ids):
+        for enc_id in sorted(encoding_ids):
+            for lang_id in sorted(language_ids):
+
+                # Check if the full name record exists
+                if name_table.getName(full_name_id, plat_id, enc_id, lang_id) is None:
+                    # The full_name_id wasn't found. No point in going further
+                    continue
+
+                # Iterate over all possible family name records
+                for family_name_id in (NameID.FONT_FAMILY_NAME,
+                                       NameID.TYPOGRAPHIC_FAMILY_NAME,
+                                       NameID.WWS_FAMILY_NAME):
+                    if name_table.getName(
+                            family_name_id, plat_id, enc_id, lang_id) is None:
+                        # The family_name_id wasn't found. Move on to the next
+                        continue
+
+                    names_compared = True  # Yay! At least one comparison will be made!!
+
+                    try:
+                        family_name = name_table.getName(
+                            family_name_id, plat_id, enc_id, lang_id).toUnicode()
+                        full_name = name_table.getName(
+                            full_name_id, plat_id, enc_id, lang_id).toUnicode()
+                    except UnicodeDecodeError as err:
+                        raise err
+
+                    if not full_name.startswith(family_name):
+                        yield FAIL, Message(
+                            "does-not",
```

Comment:
that's a weird keyword. Perhaps this one could be "mismatch"

Reply:
I'm on board with that. `does-not` is what was there before.

Reply:
Changed in e608dc2b27b105c2c31b702a2b96f060923cbe94